Applications/ConsumerApi/test/ConsumerApi.Tests.Integration/StepDefinitions/TokensStepDefinitions.cs

```diff
@@ -121,6 +132,20 @@ public async Task WhenIdentitySendsAGetRequestToTheTokensIdEndpointWithPassword(
                 : await client.Tokens.GetTokenUnauthenticated(tokenId);
     }
 
+    [When($@"{RegexFor.OPTIONAL_SINGLE_THING} sends a GET request to the /Tokens/{RegexFor.SINGLE_THING}.Id endpoint")]
+    public async Task WhenIdentitySendsAGetRequestToTheTokensIdEndpoint(string identityName, string tokenName)
+    {
+        var isAuthenticated = identityName != "-";
+
+        var client = isAuthenticated ? _clientPool.FirstForIdentityName(identityName) : _clientPool.Anonymous;
+        var tokenId = _tokensContext.CreateTokenResponses[tokenName].Id;
+
+        if (isAuthenticated)
+            _responseContext.WhenResponse = await client.Tokens.GetToken(tokenId);
+        else
+            _responseContext.WhenResponse = await client.Tokens.GetTokenUnauthenticated(tokenId);
+    }
```

Comment:
No need to support the anonymous case here.

```suggestion
    [When($@"{RegexFor.SINGLE_THING} sends a GET request to the /Tokens/{RegexFor.SINGLE_THING}.Id endpoint")]
    public async Task WhenIdentitySendsAGetRequestToTheTokensIdEndpoint(string identityName, string tokenName)
    {
        var client = _clientPool.FirstForIdentityName(identityName);
        var tokenId = _tokensContext.CreateTokenResponses[tokenName].Id;
        _responseContext.WhenResponse = await client.Tokens.GetToken(tokenId);
    }
```